core/src/commonMain/kotlin/com/powersync/sync/SyncStream.kt

```diff
@@ -96,11 +96,11 @@ internal class SyncStream(
 //                    break;
 //                }
             } catch (e: Exception) {
-                // If the coroutine was cancelled, don't log an error
-                if (e !is CancellationException) {
-                    logger.e(e) { "Error in streamingSync" }
+                if (e is CancellationException) {
+                    throw e
```

Comment:
Whats the reason for rethrowing here? How would the error surface? What would the difference be compared to breaking out of the loop?

Reply:
I'm following this guide https://medium.com/mobilepeople/how-to-break-a-coroutine-cancellation-12e968c31642 which explains that swallowing a cancellation exception may lead to unintended consequences as the coroutine will continue it's execution until the next suspend call. 

The error surfacing would depend on what resulted in the cancellation. 

The difference is that an error will be thrown with the reason for the error whereas breaking out of the loop will just break out of the loop without a reason which could lead to difficult to debug issues.

Reply:
Thanks for the context. Sounds good to me.